lib/extractSignature.js

```diff
@@ -0,0 +1,17 @@
+'use strict'
+
+const extractSignatureHeaderAsDict = require('./extractSignatureHeaderAsDict')
+
+function extractSignature (req) {
+  const signatureHeaderDict = extractSignatureHeaderAsDict(req)
+  if (signatureHeaderDict instanceof Error) {
```

Comment:
We should think a bit about how we want the error handling to work here (and in the plugin more generally). In this case we are returning an error from `extractSignatureHeaderAsDict` in certain cases, and then re-returning it.

I wonder if it would make sense to simply throw errors from inside these methods and catch them in the plugins implementation (wrap the entire extract / verify process in a try/catch, If anything goes wrong, catch the error there and handle it gracefully). I think that would be simpler than checking for errors at multiple levels and handling them that way. Thoughts?

Reply:
I certainly didn't like the error handling that I had in there, which is why this is still a draft pr. I'm thinking a structure kind of like this:
```
helpers:
extractSigHeaders => if(bad) throw error
extractSig => if(bad) throw error

plugin.js:
try {
pipe(
    res,
    extractSigHeaders,
    extractSig
    )
} catch (error) {
 do something
}
```

Reply:
☝️ Exactly